lib/index.jsx

```diff
@@ -15,9 +15,19 @@ class DatetimeRangePicker extends Component {
     this.state = {
       start: moment(props.startDate) || moment(),
       end: moment(props.endDate) || moment(),
+      startDate: moment(props.startDate) || moment(),
+      endDate: moment(props.endDate) || moment(),
     };
   }
 
+  static getDerivedStateFromProps(nextProps, prevState) {
```

Comment:
Store `nextProps.startDate` & `nextProps.endDate` in `state.propsStartDate` & `state.propsEndDate` without calling moment. This'll allow checking for changes with `===` instead of calling `moment`. Only when it changes put calculated `moment` results into `state.start` & `state.end`. Then constructor can be updated to initilate as `this.state = { start: null, end: null, startDate: null, endDate: null }`

Reply:
Also changed in: https://github.com/snamoah/react-datetime-range-picker/pull/7/commits/59b6c18c9057ff293469dde27364490b5e3014a1